yt_dlp/extractor/tvp.py

```diff
@@ -482,21 +482,32 @@ def _real_extract(self, url):
 class TVPVODBaseIE(InfoExtractor):
     _API_BASE_URL = 'https://vod.tvp.pl/api/products'
 
-    def _call_api(self, resource, video_id, **kwargs):
+    def _call_api(self, resource, video_id, query={}, **kwargs):
         return self._download_json(
             f'{self._API_BASE_URL}/{resource}', video_id,
-            query={'lang': 'pl', 'platform': 'BROWSER'}, **kwargs)
+            query={'lang': 'pl', 'platform': 'BROWSER', **query}, **kwargs)
 
-    def _parse_video(self, video):
-        return {
-            '_type': 'url',
-            'url': 'tvp:' + video['externalUid'],
-            'ie_key': TVPEmbedIE.ie_key(),
+    def _parse_video(self, video, with_url=True):
+        info_dict = {
+            'id': str(video['id']),
             'title': video.get('title'),
-            'description': traverse_obj(video, ('lead', 'description')),
+            'description': clean_html(video.get('lead') or video.get('description')),
             'age_limit': int_or_none(video.get('rating')),
             'duration': int_or_none(video.get('duration')),
+            'episode_number': int_or_none(video.get('number')),
+            'series': traverse_obj(video, ('season', 'serial', 'title')),
+            'thumbnails': [{'url': u}
+                           for u in traverse_obj(video,
+                           ('images', ..., ..., 'url'), expected_type=url_or_none)
+                           if u is not None],
         }
```

Comment:
Actually, this whole thing can be simplified with new traverse_obj features.
```suggestion
        info_dict = traverse_obj(video, {
            'id': ('id', {str_or_none}),
            'title': 'title',
            'age_limit': ('rating', {int_or_none}),
            'duration': ('duration', {int_or_none}),
            'episode_number': ('number', {int_or_none}),
            'series': ('season', 'serial', 'title', {str}),
            'thumbnails': ('images', ..., ..., {'url': ('url', {url_or_none})}),
        })
        info_dict['description'] = traverse_obj(video, 'lead', 'description', expected_type=clean_html)
```

Reply:
This was the reason I had behind wanting to discard empty string.

Alternatively we can finally introduce `truthy_or_none` and do
```py
description = traverse_obj(video, (('lead', 'description'), {clean_html}, {truthy_or_none}), get_all=False)
```

`.get()` can throw if input data is not a dict.

Reply:
> `.get()` can throw if input data is not a dict.

That's not an issue here. Ig, we can go without traverse_obj for the time being. But still `dict_get` is better than `.get` imo.

```py
info_dict['description'] = clean_html(dict_get(video, 'lead', 'description'))
```

Reply:
I think it'd need to be
```py
info_dict['description'] = clean_html(dict_get(video, ('lead', 'description')))
```

Reply:
changed